qiskit/pulse/builder.py

```diff
@@ -870,6 +957,21 @@ def _call_gate(self, gate, qargs):
 
         self._lazy_circuit.append(gate, qargs=qargs)
 
+    @staticmethod
+    def _naive_typecast_schedule(schedule: Schedule):
+        # Naively convert into ScheduleBlock
+        from qiskit.pulse.transforms import inline_subroutines, flatten, pad
+
+        preprocessed_schedule = inline_subroutines(flatten(schedule))
+        pad(preprocessed_schedule, inplace=True, pad_with=instructions.AreaBarrier)
+
+        # default to left alignment, namely ASAP scheduling
```

Comment:
This is requirements because you are converting schedule. The purpose of alignment here is "not" to break ordering of original instructions. Schedule is the representation of instructions, which is "ascending order" in time t0. This should correspond to left alignment. If you choose other alignment, you may break ordering of instructions.